src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleCreationExpression.cs

```diff
@@ -59,7 +56,18 @@ private BoundNode RewriteTupleCreationExpression(BoundTupleCreationExpression no
                                                                                               TupleTypeSymbol.RestPosition - 1)
                                                                                       .Add(currentCreation);
 
-                        MethodSymbol constructor = tuple8Ctor.AsMember(underlyingTupleTypeChain.Pop());
+                        NamedTypeSymbol currentType = underlyingTupleTypeChain.Pop();
+                        var tuple8Ctor = (MethodSymbol)TupleTypeSymbol.GetWellKnownMemberInType(currentType.OriginalDefinition,
+                                                                                                TupleTypeSymbol.GetTupleCtor(TupleTypeSymbol.RestPosition),
+                                                                                                _compilation.Assembly,
+                                                                                                _diagnostics,
+                                                                                                node.Syntax);
+                        if ((object)tuple8Ctor == null)
+                        {
+                            return node;
+                        }
```

Comment:
If the reason to put this code into the loop is the fact that you need to pop the type from the chain builder before entering the loop, you can Peek it instead, or simply get the last item in the builder.